web-common/src/features/dashboards/stores/dashboard-stores.ts

```diff
@@ -375,21 +196,14 @@ const metricViewReducers = {
     });
   },
 
-  sync(
-    name: string,
-    metricsView: V1MetricsView,
-    fullTimeRange: V1ColumnTimeRangeResponse | undefined
-  ) {
+  sync(name: string, metricsView: V1MetricsView) {
```

Comment:
I'm unclear about the separation of concerns for this `sync` function. It syncs measures and dimensions, but not time controls. Why?

Also, is this function still necessary now that you've added [this code](https://github.com/rilldata/rill/pull/3182/files#diff-e65c925c7fdd33020a3635868ebecd92ccc541aacec0a9100ae8b88b16b99aeaR41-R42)?

I might just be getting confused by the generic "sync" function name – maybe a more specific name will clear this up.

Reply:
For measures and dimensions there is a clear idea on how we can merge them with current state. EG: if a selected leaderboard measure is deleted we should select the next valid one.

There is no defined merge for time controls. We definitely need one. But just replacing everything to default will cause issues when sync is called and there is URL state.

Reply:
Lets reduce the scope of this PR to only deal with default time range issue and not the sync issue.